server/src/main/java/org/elasticsearch/ingest/IngestStats.java

```diff
@@ -223,8 +241,8 @@ Builder addTotalMetrics(IngestMetric totalMetric) {
             return this;
         }
 
-        Builder addPipelineMetrics(String pipelineId, IngestMetric pipelineMetric) {
-            this.pipelineStats.add(new PipelineStat(pipelineId, pipelineMetric.createStats()));
+        Builder addPipelineMetrics(String pipelineId, IngestMetric pipelineMetric, PipelineMetric byteMetrics) {
+            this.pipelineStats.add(new PipelineStat(pipelineId, pipelineMetric.createStats(), byteMetrics.createByteStats()));
```

Comment:
The naming here is a bit confusing with the variable of type `IngestMetric` being `pipelineMetric` and the variable of type `PipelineMetric` being `byteMetrics`. I think I understand why this is the case - `IngestMetric pipelineMetric` is named this way because `IngestMetric` can apply to processors so the variable clarifies that this is for a pipeline, while the `PipelineMetric` type is named so because it is specific to pipelines. This make sense, but I'm wondering how we can make the names clearer ... here are a couple ideas:

1. `PipelineMetric` => `PipelineByteMetric`  or `ByteMetric` - This would make the class less flexible for future pipeline specific metrics, but that's probably okay. 
2. `PipelineMetric` => `PipelineOnlyMetric` or `PipelineSpecificMetric` - Keep the class name general, but make it more clear how it differs from `IngestMetric`

What do you think?

Reply:
Yeah I wasn't quite sure how to handle the naming here so I'm glad to be having this discussion. I'm personally leaning towards `PipelineSpecificMetric`, but I'm ok with either one
I'm also thinking I probably should have changed `IngestMetric pipelineMetric` to `IngestMetric ingestMetric` in the method signature here (I'll change that in the next commit).

Reply:
Just had another thought. How about `IngestPipelineMetric` ? That way it's more clearly analogous to `IngestMetric`, but is specific to a pipeline.